src/Extension/AccountReset/SecurityExtension.php

```diff
@@ -0,0 +1,179 @@
+<?php declare(strict_types=1);
+
+namespace SilverStripe\MFA\Extension\AccountReset;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Extensible;
+use SilverStripe\Core\Extension;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\FormAction;
+use SilverStripe\Forms\PasswordField;
+use SilverStripe\Forms\RequiredFields;
+use SilverStripe\MFA\JSONResponse;
+use SilverStripe\MFA\RequestHandler\BaseHandlerTrait;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\ORM\ValidationResult;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\Security;
+
+/**
+ * Extends the Security controller to support Account Resets. This extension can
+ * itself be extended to add procedures to the reset action (such as removing
+ * additional authentication factors, sending alerts, etc.)
+ *
+ * @package SilverStripe\MFA\Extension
+ * @property Security owner
+ */
+class SecurityExtension extends Extension
+{
+    use BaseHandlerTrait;
+    use Extensible;
+    use JSONResponse;
+
+    private static $url_handlers = [
+        'GET reset-account' => 'resetaccount',
+    ];
+
+    private static $allowed_actions = [
+        'resetaccount',
+        'ResetAccountForm',
+    ];
+
+    public function resetaccount(HTTPRequest $request): HTTPResponse
+    {
+        if (Security::getCurrentUser()) {
+            return $this->jsonResponse(['error' => 'Already authenticated'], 400);
+        }
+
+        ['m' => $memberID, 't' => $token] = $request->getVars();
```

Comment:
This syntax confuses me, what's it doing?

Reply:
Yeah this is new in PHP 7.1.